compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/utils/Utils.java

```diff
@@ -221,7 +221,7 @@ private static String getDataStoreName(Path configPath) throws BalException {
                     }
                 }
             }
-            throw new BalException("the persist.datastore configuration does not exist in the Ballerina.toml file");
+            return null;
```

Comment:
```suggestion
            // Skip data store-specific validations if the `tool.persist` configuration does not exist in the `toml` files, instead of returning an error.
            return null;
```